libs/community/langchain_community/vectorstores/falkordb_vector.py

```diff
@@ -0,0 +1,1684 @@
+        username: Optional[str] = None,
+        password: Optional[str] = None,
+        host: str = "localhost",
+        port: int = 6379,
+        distance_strategy: DistanceStrategy = DEFAULT_DISTANCE_STRATEGY,
+        database: str = generate_random_string(4),
+        node_label: str = "Chunk",
+        relation_type: str = "",
+        embedding_node_property: str = "embedding",
+        text_node_property: str = "text",
+        embedding_dimension: Optional[int] = None,
+        retrieval_query: str = "",
+        index_type: IndexType = DEFAULT_INDEX_TYPE,
+        graph: Optional[FalkorDBGraph] = None,
+        relevance_score_fn: Optional[Callable[[float], float]] = None,
+        ssl: bool  = False,
+        pre_delete_collection: bool = False,
+    ) -> None:
+        try:
+            import falkordb 
+        except ImportError:
+            raise ImportError(
+                "Could not import falkordb python package."
+                "Please install it with `pip install falkordb`"
+            )
+        
+        # Only euclidean distance strategies is avaliable in falkor for now
+        if distance_strategy != DistanceStrategy.EUCLIDEAN_DISTANCE:
+            raise ValueError(
+                "distance_strategy must be 'EUCLIDEAN_DISTANCE'"
+            )
+        
+        # Graph object takes precedent over env or input params
+        if graph:
+            self._database = graph._graph
+            self._driver = graph._driver
+        else:
+            # Handle credentials via environment variables or input params
+            self._host = host
+            self._port = port
+            self._username = username or os.environ.get("FALKORDB_USERNAME")
+            self._password = password or os.environ.get("FALKORDB_PASSWORD")
+            self._ssl = ssl
+
+            # Initialize the FalkorDB connection
+            # Initialize the FalkorDB connection
+            try:
+                self._driver = falkordb.FalkorDB(
+                    host=self._host,
+                    port=self._port,
+                    username=self._username,
+                    password=self._password,
+                    ssl=self._ssl,
+                )
+            except redis.exceptions.ConnectionError:
+                raise ValueError(
+                    "Could not connect to FalkorDB database."
+                    "Please ensure that the host and port is correct"
+                )
+            except redis.exceptions.AuthenticationError:
+                raise ValueError(
+                    "Could not connect to FalkorDB database. "
+                    "Please ensure that the username and password are correct"
+                )
+            
+            # Verify that required values are not null
+            check_if_not_null(
+                ["embedding_node_property","node_label"],
+                [embedding_node_property, node_label],
+            )
+
+            print("Database name: ", database)
```

Comment:
why use print?

Reply:
This particular line of code is for me and my debugging. I was supposed to remove it before pushing. I wanted to know which database I was working with whenever I initializing a new vectorstore